test_programs/compile_failure/unspecified_generic/src/main.nr

```diff
@@ -0,0 +1,5 @@
+fn foo<T>() {}
+
+fn main() {
+    foo();
```

Comment:
I think it is better to move error cases under `noirc_frontend/tests.rs` when we can. This is non-blocking though, just taking note.

Reply:
Can't move it to the frontend tests unfortunately since this error is during monomorphization.